python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/aaa-server-groups.j2

```diff
@@ -4,22 +4,20 @@
  that can be found in the LICENSE file.
 #}
 {# eos - aaa server groups #}
-{% if aaa_server_groups is arista.avd.defined %}
-{%     for aaa_server_group in aaa_server_groups %}
-{%         if aaa_server_group.type is arista.avd.defined and aaa_server_group.name is arista.avd.defined %}
+{% for aaa_server_group in aaa_server_groups | arista.avd.natural_sort('type') | arista.avd.natural_sort('name') %}
+{%     if aaa_server_group.type is arista.avd.defined and aaa_server_group.name is arista.avd.defined %}
 !
 aaa group server {{ aaa_server_group.type }} {{ aaa_server_group.name }}
-{%             if aaa_server_group.servers is arista.avd.defined %}
-{%                 for server in aaa_server_group.servers %}
-{%                     if server.server is arista.avd.defined %}
-{%                         set server_cli = "server " ~ server.server %}
-{%                         if server.vrf is arista.avd.defined %}
-{%                             set server_cli = server_cli ~ " vrf " ~ server.vrf %}
-{%                         endif %}
-   {{ server_cli }}
+{%         if aaa_server_group.servers is arista.avd.defined %}
+{%             for server in aaa_server_group.servers %}
```

Comment:
should the servers also be sorted somehow?

Reply:
EOS is not sorting the servers. It is just appending the new entry.